skyportal/tests/fixtures.py

```diff
@@ -530,26 +532,56 @@ def teardown(obj):
         DBSession().delete(obj)
         DBSession().commit()
         for instrument in instruments:
-            InstrumentFactory.teardown(instrument)
+            InstrumentFactory.teardown(instrument.id)
 
 
-class GcnFactory(factory.alchemy.SQLAlchemyModelFactory):
+class GcnNoticeFactory(factory.alchemy.SQLAlchemyModelFactory):
     class Meta(BaseMeta):
-        model = GcnEvent
+        model = GcnNotice
 
     sent_by = factory.SubFactory(UserFactory)
 
     dateobs = datetime.datetime.now()
 
+    ivorn = str(uuid.uuid4())
+
     @staticmethod
-    def teardown(gcn):
-        if is_already_deleted(gcn, GcnEvent):
-            return
+    def teardown(gcnnotice_id):
+        gcnnotice_ = (
+            DBSession()
+            .execute(sa.select(GcnNotice).filter(GcnNotice.id == gcnnotice_id))
+            .scalars()
+            .first()
+        )
+        if gcnnotice_ is not None:
+            # If it is, delete it
+            DBSession().delete(gcnnotice_)
+            DBSession().commit()
 
-        sent_by = gcn.sent_by.id
-        DBSession().delete(gcn)
-        DBSession().commit()
-        UserFactory.teardown(sent_by)
+
+class GcnEventFactory(factory.alchemy.SQLAlchemyModelFactory):
+    class Meta(BaseMeta):
+        model = GcnEvent
+
+    sent_by = factory.SubFactory(UserFactory)
+
+    dateobs = datetime.datetime.now()
+
+    @staticmethod
+    def teardown(gcnevent_id):
+        gcnevent_ = (
+            DBSession()
+            .execute(sa.select(GcnEvent).filter(GcnEvent.id == gcnevent_id))
+            .scalars()
+            .first()
+        )
+        if gcnevent_ is not None:
+            # If it is, delete it
+            gcn_notices = gcnevent_.gcn_notices
+            for gcn_notice in gcn_notices:
+                GcnNoticeFactory.teardown(gcn_notice.id)
+            DBSession().delete(gcnevent_)
```

Comment:
same here

Reply:
So maybe we need to build the session and pass it in for teardown?

Reply:
Not necessarily. If the teardown is a logical unit, that doesn't need to get objects connected to a session (you are giving IDs not objects) then just starting a new session inside the function is a good way. Just turn it into a context manager when the function starts, so you don't leave an open session. 
You could get into trouble if you already opened a session outside, and then DBSession grabs it, and then the context manager closes that in the teardown. That's why we need to move the whole concept of session to be more localized, so these sort of things don't happen.